pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -176,6 +176,16 @@ func (c *Cluster) Start(clusterInfo *cephconfig.ClusterInfo, rookVersion string,
 		return nil, fmt.Errorf("failed to initialize ceph cluster info. %+v", err)
 	}
 
+	nodeList, err := c.getFullNodeList()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get node list. %+v", err)
+	}
+	validNodes := c.getValidNodes(nodeList.Items)
+	nodeCount := len(validNodes)
+	if !c.spec.Mon.AllowMultiplePerNode && c.spec.Mon.Count > nodeCount {
+		return nil, fmt.Errorf("the desired mon count is %d while the number of available nodes is %d despite allowMultiplePerNode is %t", c.spec.Mon.Count, nodeCount, c.spec.Mon.AllowMultiplePerNode)
```

Comment:
The node check looks correct. Now I'm wondering where the right place is to return the error. If a cluster is already running with the correct number of mons, we should allow the cluster to continue running. 

Where we want this check is when adding a new mon. If we try to add a new mon, we should fail if there aren't enough nodes. This would apply both to adding a new mon when creating the new cluster, or during the health check when we are failing over a mon.

Reply:
I agree with you. Unless the number of new mons exceeds the number of available nodes when assigning it, multiple mons will not be scheduled on the same node.
I’ll modify that to check when assigning.